document/workflow/variable-tree/data-source/skeleton-tree-data-source.ts

```diff
@@ -550,7 +561,10 @@ export class SkeletonTreeDataSource implements TreeNodeFactory, TreeNodeObserver
      * Clears all `marked` states of all nodes
      */
     clearMarks() {
-        this.root$.pipe(first()).subscribe(root => root.unmarkRecursively());
+        this.root$.pipe(
+            filter(root => !!root),
+            first()
```

Comment:
Better 
this.root$.pipe(
            first()
            filter(root => !!root),

Reply:
In your case, the subscription wouldn't be called for `root` being *null* at first. What I want is the first emit of `root` being not *null*.
I saw you can put a predicate inside of `first`. So `pipe(first(root => !!root))` is even better.